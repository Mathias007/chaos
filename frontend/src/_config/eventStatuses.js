const eventStatuses = {
    articles: {
        ARTICLE_SUCCESSFULLY_LOADED: "ARTICLE_SUCCESSFULLY_LOADED",
        ARTICLE_NOT_FOUND: "ARTICLE_NOT_FOUND",
        ARTS_LIST_LOADED: "ARTS_LIST_LOADED",
        ARTS_LIST_NOT_FOUND: "ARTS_LIST_NOT_FOUND",
        ARTICLE_ADDED: "ARTICLE_ADDED",
        ARTICLE_ADDING_FAILED: "ARTICLE_ADDING_FAILED",
        ARTICLE_SUCCESFULLY_EDITED: "ARTICLE_SUCCESFULLY_EDITED",
        ARTICLE_EDITING_FAILED: "ARTICLE_EDITING_FAILED",
        ARTICLE_SUCCESFULLY_DELETED: "ARTICLE_SUCCESFULLY_DELETED",
        ARTICLE_DELETING_FAILED: "ARTICLE_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    },
    auth: {
        PASSWORD_TOO_SHORT: "PASSWORD_TOO_SHORT",
        PASSWORD_INCORRECT: "PASSWORD_INCORRECT",
        PASSWORDS_NOT_MATCH: "PASSWORDS_NOT_MATCH",
        USER_LOADED: "USER_LOADED",
        LOGIN_SUCCESSFUL: "LOGIN_SUCCESSFUL",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        LOGIN_FAILED: "LOGIN_FAILED",
        LOGOUT_SUCCESSFUL: "LOGOUT_SUCCESSFUL",
        RESET_STATUS: "RESET_STATUS"
    },
    news: {
        NEWS_SUCCESSFULLY_LOADED: "NEWS_SUCCESSFULLY_LOADED",
        NEWS_NOT_FOUND: "NEWS_NOT_FOUND",
        NEWS_LIST_LOADED: "NEWS_LIST_LOADED",
        NEWS_LIST_NOT_FOUND: "NEWS_LIST_NOT_FOUND",
        NEWS_ADDED: "NEWS_ADDED",
        NEWS_ADDING_FAILED: "NEWS_ADDING_FAILED",
        NEWS_SUCCESFULLY_EDITED: "NEWS_SUCCESFULLY_EDITED",
        NEWS_EDITING_FAILED: "NEWS_EDITING_FAILED",
        NEWS_SUCCESFULLY_DELETED: "NEWS_SUCCESFULLY_DELETED",
        NEWS_DELETING_FAILED: "NEWS_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    },
    knights: {
        KAM_BUILDINGS_LOADED: "KAM_BUILDINGS_LOADED",
        KAM_BUILDINGS_NOT_FOUND: "KAM_BUILDINGS_NOT_FOUND",
        KAM_UNITS_LOADED: "KAM_UNITS_LOADED",
        KAM_UNITS_NOT_FOUND: "KAM_UNITS_NOT_FOUND",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR"
    },
    tzar: {
        TZAR_NATIONS_LOADED: "TZAR_NATIONS_LOADED",
        TZAR_NATIONS_NOT_FOUND: "TZAR_NATIONS_NOT_FOUND",
        TZAR_UNITS_LOADED: "TZAR_UNITS_LOADED",
        TZAR_UNITS_NOT_FOUND: "TZAR_UNITS_NOT_FOUND",
        TZAR_SPELLS_LOADED: "TZAR_SPELLS_LOADED",
        TZAR_SPELLS_NOT_FOUND: "TZAR_SPELLS_NOT_FOUND",
        TZAR_TECHNOLOGIES_LOADED: "TZAR_TECHNOLOGIES_LOADED",
        TZAR_TECHNOLOGIES_NOT_FOUND: "TZAR_TECHNOLOGIES_NOT_FOUND",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR"
    },
    users: {
        USER_SUCCESSFULLY_LOADED: "USER_SUCCESSFULLY_LOADED",
        USER_NOT_FOUND: "USER_NOT_FOUND",
        USERS_LIST_LOADED: "USERS_LIST_LOADED",
        USERS_LIST_NOT_FOUND: "USERS_LIST_NOT_FOUND",
        USER_ADDED: "USER_ADDED",
        USER_ADDING_FAILED: "USER_ADDING_FAILED",
        USER_SUCCESFULLY_EDITED: "USER_SUCCESFULLY_EDITED",
        USER_EDITING_FAILED: "USER_EDITING_FAILED",
        USER_SUCCESFULLY_DELETED: "USER_SUCCESFULLY_DELETED",
        USER_DELETING_FAILED: "USER_DELETING_FAILED",
        AUTHENTICATION_ERROR: "AUTHENTICATION_ERROR",
        RESET_STATUS: "RESET_STATUS"
    }
};

export default eventStatuses;
