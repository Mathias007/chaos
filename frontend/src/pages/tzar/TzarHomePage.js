import React, { Component } from "react";
import { Layout } from "antd";

import HeaderComponent from "../global/HeaderComponent";
import SidebarComponent from "../global/SidebarComponent";
import FooterComponent from "../global/FooterComponent";

import TzarHomeContent from "./components/TzarHomeContent";

class TzarHomePage extends Component {
    state = {};
    render() {
        return (
            <div className="App-container">
                <Layout>
                    <HeaderComponent />
                    <Layout>
                        <SidebarComponent />
                        <TzarHomeContent />
                    </Layout>
                </Layout>
                <FooterComponent />
            </div>
        );
    }
}

export default TzarHomePage;