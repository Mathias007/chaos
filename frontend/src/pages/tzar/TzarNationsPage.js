import React, { Component } from "react";
import { Layout } from "antd";

import GlobalPageHeader from "../components/GlobalPageHeader";
import GlobalSidebar from "../components/GlobalSidebar";
import GlobalPageFooter from "../components/GlobalPageFooter";

import TzarNationsContent from "./components/TzarNationsContent";

class TzarNationsPage extends Component {
    state = {};
    render() {
        return (
            <div className="App-container">
                <Layout>
                    <GlobalPageHeader />
                    <Layout>
                        <GlobalSidebar />
                        <TzarNationsContent />
                    </Layout>
                </Layout>
                <GlobalPageFooter />
            </div>
        );
    }
}

export default TzarNationsPage;