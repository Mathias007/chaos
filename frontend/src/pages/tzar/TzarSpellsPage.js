import React, { Component } from "react";

import allocationKeys from "../../_config/allocationKeys";
import navigationTitles from "../../_config/navigationTitles";

import BreadcrumbComponent from "../global/BreadcrumbComponent";
import PageHeaderComponent from "../components/PageHeaderComponent";
import PageContentComponent from "../components/PageContentComponent";
import TzarSpellsTable from "./data/TzarSpellsTable";

import { Layout } from "antd";
import styles from "../../styles/styles";

const { TZAR_BURDEN_CROWN } = navigationTitles;
const { TZAR_SPELLS } = allocationKeys;

class TzarSpellsPage extends Component {
    state = {
        allocationKey: TZAR_SPELLS
    };
    render() {
        return (
            <Layout style={styles.layout}>
                <BreadcrumbComponent
                    section={TZAR_BURDEN_CROWN}
                    page={navigationTitles.TZAR_SPELLS}
                />
                <PageHeaderComponent allocationKey={this.state.allocationKey} />
                <PageContentComponent
                    allocationKey={this.state.allocationKey}
                />
                <TzarSpellsTable />
            </Layout>
        );
    }
}

export default TzarSpellsPage;
