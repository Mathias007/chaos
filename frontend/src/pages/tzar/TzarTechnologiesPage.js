import React, { Component } from "react";

import allocationKeys from "../../_config/allocationKeys";

import BreadcrumbComponent from "../global/BreadcrumbComponent";
import PageHeaderComponent from "../components/PageHeaderComponent";
import PageContentComponent from "../components/PageContentComponent";

import TzarTechnologiesTable from "./data/TzarTechnologiesTable";
import { Layout } from "antd";

const { TZAR_TECHNOLOGIES } = allocationKeys;

class TzarTechnologiesPage extends Component {
    state = {
        allocationKey: TZAR_TECHNOLOGIES
    };
    render() {
        return (
            <Layout style={{ padding: "0 24px 24px" }}>
                <BreadcrumbComponent />
                <PageHeaderComponent allocationKey={this.state.allocationKey} />
                <PageContentComponent
                    allocationKey={this.state.allocationKey}
                />
                <TzarTechnologiesTable />
            </Layout>
        );
    }
}

export default TzarTechnologiesPage;
