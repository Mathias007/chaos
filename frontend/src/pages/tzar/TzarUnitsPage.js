import React from "react";

import allocationKeys from "../../_config/allocationKeys";
import navigationTitles from "../../_config/navigationTitles";

import styles from "../../styles/styles";

import BreadcrumbComponent from "../global/BreadcrumbComponent";
import PageHeaderComponent from "../components/PageHeaderComponent";
import PageContentComponent from "../components/PageContentComponent";
import UnitsTable from "./data/UnitsTable";

import { Layout } from "antd";

const { TZAR_BURDEN_CROWN } = navigationTitles;
const { TZAR_UNITS } = allocationKeys;

export default function TzarUnitsPage(props) {
    return (
        <Layout style={styles.layout}>
            <BreadcrumbComponent
                section={TZAR_BURDEN_CROWN}
                page={navigationTitles.TZAR_UNITS}
            />
            <PageHeaderComponent allocationKey={TZAR_UNITS} />
            <PageContentComponent allocationKey={TZAR_UNITS} />
            <UnitsTable />
        </Layout>
    );
}
