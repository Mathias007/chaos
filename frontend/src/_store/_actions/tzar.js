import eventStatuses from "../../_config/eventStatuses";

import {fetchNationsList, fetchSpellsList, fetchTechnologiesList, fetchUnitsList  } from "./services/TzarService";

const {
    TZAR_NATIONS_LOADED,
    TZAR_UNITS_LOADED,
    TZAR_SPELLS_LOADED,
    TZAR_TECHNOLOGIES_LOADED
} = eventStatuses.tzar;

export const showNations = () => {
    return (dispatch, getState) => {
        const dispatchNationsListLoaded = function(nations) {
            dispatch({
                type: TZAR_NATIONS_LOADED,
                data: nations
            });
        };

        return fetchNationsList(dispatchNationsListLoaded);
    };
};

export const showUnits = () => {
    return (dispatch, getState) => {
        const dispatchUnitsListLoaded = function(units) {
            dispatch({
                type: TZAR_UNITS_LOADED,
                data: units
            });
        };

        return fetchUnitsList(dispatchUnitsListLoaded);
    };
};

export const showSpells = () => {
    return (dispatch, getState) => {
        const dispatchSpellsListLoaded = function(spells) {
            dispatch({
                type: TZAR_SPELLS_LOADED,
                data: spells
            });
        };

        return fetchSpellsList(dispatchSpellsListLoaded);
    };
};

export const showTechnologies = () => {
    return (dispatch, getState) => {
        const dispatchTechnologiesListLoaded = function(technologies) {
            dispatch({
                type: TZAR_TECHNOLOGIES_LOADED,
                data: technologies
            });
        };

        return fetchTechnologiesList(dispatchTechnologiesListLoaded);
    };
};
