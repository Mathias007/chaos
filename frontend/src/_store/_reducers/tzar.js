import eventStatuses from "../../_config/eventStatuses";

const {
    TZAR_NATIONS_LOADED,
    TZAR_NATIONS_NOT_FOUND,
    TZAR_UNITS_LOADED,
    TZAR_UNITS_NOT_FOUND,
    TZAR_SPELLS_LOADED,
    TZAR_SPELLS_NOT_FOUND,
    TZAR_TECHNOLOGIES_LOADED,
    TZAR_TECHNOLOGIES_NOT_FOUND,
    AUTHENTICATION_ERROR
} = eventStatuses.tzar;

const initialState = {
    nations: [],
    spells: [],
    technologies: [],
    units: [],
    errorMessage: ""
};

export default function tzar(state = initialState, action) {
    switch (action.type) {
        case TZAR_NATIONS_LOADED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                nations: action.data.nations,
                errorMessage: action.data.message
            };

        case TZAR_NATIONS_NOT_FOUND:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case TZAR_SPELLS_LOADED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                spells: action.data.spells,
                errorMessage: action.data.message
            };

        case TZAR_SPELLS_NOT_FOUND:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case TZAR_TECHNOLOGIES_LOADED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                technologies: action.data.technologies,
                errorMessage: action.data.message
            };

        case TZAR_TECHNOLOGIES_NOT_FOUND:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case TZAR_UNITS_LOADED:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                units: action.data.units,
                errorMessage: action.data.message
            };

        case TZAR_UNITS_NOT_FOUND:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        case AUTHENTICATION_ERROR:
            console.log(action.data);
            return {
                ...state,
                ...action.data,
                errorMessage: action.data.message
            };

        default:
            return state;
    }
}
