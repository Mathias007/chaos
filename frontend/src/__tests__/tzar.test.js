import eventStatuses from "../_config/eventStatuses";
import tzar from "../_store/_reducers/tzar";

const {
    TZAR_NATIONS_LOADED,
    TZAR_NATIONS_NOT_FOUND,
    TZAR_UNITS_LOADED,
    TZAR_UNITS_NOT_FOUND,
    TZAR_SPELLS_LOADED,
    TZAR_SPELLS_NOT_FOUND,
    TZAR_TECHNOLOGIES_LOADED,
    TZAR_TECHNOLOGIES_NOT_FOUND,
    AUTHENTICATION_ERROR
} = eventStatuses.tzar;

const initialState = {
    nations: [],
    spells: [],
    technologies: [],
    units: [],
    errorMessage: ""
};

describe("test users reducer", () => {
    const tzarList = {
        message: "Pozytywna wiadomość z serwera!",
        data: [
            {
                _id: "1111-2222-3333-4444",
                name: "Hobok",
                cost: "1488 gold",
                hp: "500",
                worker: "tester"
            }
        ]
    };

    const serverMessage = { message: "Ważna wiadomość z serwera!" };

    it("should return the initial state", () => {
        expect(tzar(undefined, {})).toEqual(initialState);
    });

    it("should load nations list", () => {
        const action = {
            type: TZAR_NATIONS_LOADED,
            data: tzarList.data
        };
        expect(tzar(initialState, action)).toMatchSnapshot();
    });

    it("should get message about nations list's not finding", () => {
        const action = { type: TZAR_NATIONS_NOT_FOUND, data: serverMessage };
        expect(tzar(initialState, action)).toMatchSnapshot();
    });

    it("should load spells list", () => {
        const action = {
            type: TZAR_SPELLS_LOADED,
            data: tzarList.data
        };
        expect(tzar(initialState, action)).toMatchSnapshot();
    });

    it("should get message about nations list's not finding", () => {
        const action = { type: TZAR_SPELLS_NOT_FOUND, data: serverMessage };
        expect(tzar(initialState, action)).toMatchSnapshot();
    });

    it("should load technologies list", () => {
        const action = {
            type: TZAR_TECHNOLOGIES_LOADED,
            data: tzarList.data
        };
        expect(tzar(initialState, action)).toMatchSnapshot();
    });

    it("should get message about nations list's not finding", () => {
        const action = {
            type: TZAR_TECHNOLOGIES_NOT_FOUND,
            data: serverMessage
        };
        expect(tzar(initialState, action)).toMatchSnapshot();
    });

    it("should load units list", () => {
        const action = { type: TZAR_UNITS_LOADED, data: tzarList.data };
        expect(tzar(initialState, action)).toMatchSnapshot();
    });

    it("should get message about units list's not finding", () => {
        const action = { type: TZAR_UNITS_NOT_FOUND, data: serverMessage };
        expect(tzar(initialState, action)).toMatchSnapshot();
    });

    it("should get message about a problem with authentication", () => {
        const action = { type: AUTHENTICATION_ERROR, data: serverMessage };
        expect(tzar(initialState, action)).toMatchSnapshot();
    });
});
