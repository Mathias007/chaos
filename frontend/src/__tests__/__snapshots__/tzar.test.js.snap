// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test users reducer should get message about a problem with authentication 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "nations": Array [],
  "spells": Array [],
  "technologies": Array [],
  "units": Array [],
}
`;

exports[`test users reducer should get message about nations list's not finding 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "nations": Array [],
  "spells": Array [],
  "technologies": Array [],
  "units": Array [],
}
`;

exports[`test users reducer should get message about nations list's not finding 2`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "nations": Array [],
  "spells": Array [],
  "technologies": Array [],
  "units": Array [],
}
`;

exports[`test users reducer should get message about nations list's not finding 3`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "nations": Array [],
  "spells": Array [],
  "technologies": Array [],
  "units": Array [],
}
`;

exports[`test users reducer should get message about units list's not finding 1`] = `
Object {
  "errorMessage": "Ważna wiadomość z serwera!",
  "message": "Ważna wiadomość z serwera!",
  "nations": Array [],
  "spells": Array [],
  "technologies": Array [],
  "units": Array [],
}
`;

exports[`test users reducer should load nations list 1`] = `
Object {
  "errorMessage": "Pozytywna wiadomość z serwera!",
  "message": "Pozytywna wiadomość z serwera!",
  "nations": Array [
    Object {
      "_id": "1111-2222-3333-4444",
      "name": "Nacje Tzar",
    },
  ],
  "spells": Array [],
  "technologies": Array [],
  "units": Array [],
}
`;

exports[`test users reducer should load spells list 1`] = `
Object {
  "errorMessage": "Pozytywna wiadomość z serwera!",
  "message": "Pozytywna wiadomość z serwera!",
  "nations": Array [],
  "spells": Array [
    Object {
      "_id": "1111-2222-3333-4444",
      "name": "Zaklęcia Tzar",
    },
  ],
  "technologies": Array [],
  "units": Array [],
}
`;

exports[`test users reducer should load technologies list 1`] = `
Object {
  "errorMessage": "Pozytywna wiadomość z serwera!",
  "message": "Pozytywna wiadomość z serwera!",
  "nations": Array [],
  "spells": Array [],
  "technologies": Array [
    Object {
      "_id": "1111-2222-3333-4444",
      "name": "Technologie Tzar",
    },
  ],
  "units": Array [],
}
`;

exports[`test users reducer should load units list 1`] = `
Object {
  "errorMessage": "Pozytywna wiadomość z serwera!",
  "message": "Pozytywna wiadomość z serwera!",
  "nations": Array [],
  "spells": Array [],
  "technologies": Array [],
  "units": Array [
    Object {
      "_id": "1111-2222-3333-4444",
      "name": "Jednostki Tzar",
    },
  ],
}
`;
